src/SharedUtilities/models.jl

```diff
@@ -100,21 +117,39 @@ function make_update_aux(model::AbstractModel)
 end
 
 """
-    make_ode_function(model::AbstractModel)
+    make_imp_tendency(model::AbstractModel)
+
+Returns an `imp_tendency` that updates auxiliary variables and
+updates the prognostic state of variables that are stepped implicitly.
+
+`compute_imp_tendency!` should be compatible with OrdinaryDiffEq.jl solvers.
+"""
+function make_imp_tendency(model::AbstractModel)
```

Comment:
maybe we could discuss these defaults a little more. Just briefly speaking with Dennis, he pointed out that even if the implicit tendency only updates the aux state, we'll still be passing to the timestepper. it would be an unecessary update aux (which we discussed before)  - in fact it would call it every stage, not once per step, which might be less OK with us in terms of unecessary calls.

what to we pass to the ODEFunction if we really want no implicit tendency is `nothing`
```
CTS.ClimaODEFunction(
            T_exp! = explicit_tendency!,
            T_imp! = nothing,
            dss! = ClimaLSM.Soil.dss!,
```

[reminder for future PR to move the dss! function ClimaLSM.Domains?]

Reply:
one idea: we could add a boolean field to `model` like "has_implicit" and if false, imp_tendency is nothing, and if true, it does what we have here

Reply:
After discussing offline, @kmdeck and I decided to add two subtypes to our `AbstractModel` type: `AbstractImexModel` and `AbstractExpModel`. We can dispatch off of these types to define defaults that better suit the desired behavior - `imp_tendency!` will be nothing for an `AbstractExplicitModel`, but will call `update_aux!` and `compute_imp_tendency!` for an `AbstractImplicitModel`. This will allow us to take advantage of multiple dispatch, and avoid unnecessarily defining an unused `make_compute_implicit_tendency` for `AbstractExplicitModel` types (which will also give a helpful method/type error if `make_compute_implicit_tendency` is called incorrectly).